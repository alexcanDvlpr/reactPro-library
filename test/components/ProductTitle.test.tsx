import React from "react";
import renderer from "react-test-renderer";
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from "../data/Products";

describe("ProductTitle", () => {

    it("Debe mostrar el componente correctamente con el titulo personalizado", () => {
        const wrapper = renderer.create(
            <ProductTitle title="Title Test" />
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });


    it("Debe mostrar el componente correctamente con el titulo del producto (Provider)", () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

});