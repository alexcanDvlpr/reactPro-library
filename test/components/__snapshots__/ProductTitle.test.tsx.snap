// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe mostrar el componente correctamente con el titulo del producto (Provider) 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;

exports[`ProductTitle Debe mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Title Test
</span>
`;
